### ISSUE TITLE
How to download a ms-building id stac?

### ISSUE THREAD

**Issue by aborruso** [stac-utils/stac-asset#39]
Hi,
first of all thank you for this project.

As I have writter [here](https://github.com/stac-utils/pystac-client/issues/565#issuecomment-1644024933), I would like to download a ms-building id stack.
And I have tried to do it running:

```
stac-client search https://planetarycomputer.microsoft.com/api/stac/v1 -c ms-buildings --bbox 46.40 24.52 47.110 25.165 --max-items 1 | stac-asset download
```

But I have this error:

```
stac_asset.errors.DownloadError: error when downloading asset 'data' with href 'abfs://footprints/delta/2023-04-25/ml-buildings.parquet/RegionName=KingdomofSaudiArabia/quadkey=123022211': cannot read a file with the filesystem client if it has a url scheme: abfs://footprints/delta/2023-04-25/ml-buildings.parquet/RegionName=KingdomofSaudiArabia/quadkey=123022211
```

Is there a way to download it?

**Comment by bmcandr** [comment]
It looks like the `abfs://` scheme is not supported yet. As shown [here](https://github.com/stac-utils/stac-asset/blob/main/src/stac_asset/planetary_computer_client.py#L46), the `PlanetaryComputerClient` is a subclass of `HttpClient` and can therefore only download assets with http-based HREFs.

**Comment by gadomski** [comment]
Yup, thanks for the report. There's (at least) two things going on here:
- [ ] We should error when we encounter unexpected protocols, instead of naïvely using `HttpClient`
- [ ] We should add support for `abfs` protocols, following some of the document hints from https://planetarycomputer.microsoft.com/docs/quickstarts/stac-geoparquet/ (e.g. `table:storage_options`)
